WIP: TASK: Add package:// stream-wrapper that is relative to the package-path to make the autoinclusion from there cleaner

mficzel opened the pull request:
**Review instructions**

The actual package:// stream wrapper would be moved to flow before merging. I added this here to have all things in one place during development.

For now the stream wrapper is also a almost identical to the resource:// stream wrapper from flow minus the option to include resources via identifier and not allowing "/../" in pathes

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

mficzel committed:
TASK: Use package:// stream-wrapper that is relative to the package-path to make the autoinclusion from there cleaner

mhsdesign commented on Neos.Neos/Classes/Domain/Service/FusionService.php:
```diff
@@ -257,15 +243,14 @@ protected function getFusionIncludes(array $fusionResources)
-                $autoIncludeNodeTypesFolder = sprintf($this->autoIncludeNodeTypesFolderPattern, $packageKey);
-                if (is_dir($autoIncludeNodeTypesFolder)) {
-                    $autoIncludeFusion[] = $autoIncludeNodeTypesFolder . '/**/*.fusion';
+                if (is_dir($package->getPackagePath() . '/NodeTypes')) {
```
i dont really like the mixture of `getPackagePath()` and `package://` side by side in this FusionService

i think you did this for performance reasons?

but either way -> can later be refactored - or not :P

mhsdesign reviewed (commented):
Nice should work conceptionally ^^

i didnt check the PackageStreamWrapper :P

mhsdesign commented:
as flows objective is to only have one way to do things we should prepare for an argument of 

`resource://My.Package/Private/Foo.html`
vs

`package://My.Package/Resource/Private/Foo.html`

mficzel commented:
Adjusted and extracted the flow part to https://github.com/neos/flow-development-collection/pull/2919 ... had to adjust the folder skipping in the filePattern resolver as the PackageStream wrapper will throw even if you simply ask wether a `..` exists.
